src/OptimizedKernelsAvx2.cc

```diff
@@ -191,6 +192,77 @@ void transpose_8rows(
         _mm256_extract_epi64(y7, 2);
     *reinterpret_cast<int64_t*>(dst + (j + 31) * ld_dst) =
         _mm256_extract_epi64(y7, 3);
+#else
+    *reinterpret_cast<int32_t*>(dst + (j + 1) * ld_dst) =
+        _mm256_extract_epi32(y0, 1);
```

Comment:
You need to implement _mm256_extract_epi64 with multiple _mm256_extract_epi32 instructions.

*reinterpret_cast<int32_t*>(dst + (j + 1) * ld_dst) =
        _mm256_extract_epi32(y0, 2);
*reinterpret_cast<int32_t*>(dst + (j + 1) * ld_dst + 4) =
        _mm256_extract_epi32(y0, 3);

Please apply a similar to the remaining part.

BTW, we don't plan to support 32-bit architecture and IIRC PyTorch also doesn't officially support it.
So even if it works we're not going to merge the PR. Just wanted to be clear up front.

Reply:
Hi @jspark1105 
 
Thanks for your suggestion.  
I change from:
reinterpret_cast<int32_t>(dst + (j + 1) * ld_dst + 4) =
_mm256_extract_epi32(y0, 3);
to:
reinterpret_cast<int32_t>(dst + (j + 1) * ld_dst + 1) =
_mm256_extract_epi32(y0, 3);

But forward quantized model in 32bit fail too,  i think there are some compatibilities and will continue debug.

Thanks anyway!